### PR TITLE
Fix numpy pinning

### PR DESCRIPTION
**Pull request by joaander**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes https://github.com/MDAnalysis/mdanalysis/issues/3660

Remove numpy > 1.18.0 specification. This causes conda-smithy to remove the numpy pinning from the build, resulting in a build against the latest numpy ABI. After this change, conda-smith rerenders the build with the proper pinning.

### CODE REVIEW

**Comment by IAlibay** [comment]
Thanks for this PR, I'm not sure removing the 1.18.0 spec is the right way to go here, I'm planning to experiment on this further later today.

**Comment by joaander** [comment]
> Thanks for this PR, I'm not sure removing the 1.18.0 spec is the right way to go here, I'm planning to experiment on this further later today.

Alright. The conda-forge gitter channel is a good resource for help sorting out pinning issues.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,14 +24,14 @@ requirements:
...
-    - numpy >=1.18.0
+    - numpy
     - cython
     - packaging
 
   run:
     - python
     - scipy >=1.0.0
-    - numpy >1.18.0
+    - numpy
```
Comment (IAlibay) [inline recipe/meta.yaml:34]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

**Review by IAlibay — CHANGES_REQUESTED** [review]
Can we try this change suggested by @mikemhenry ? From discussions it probably will have the same behaviour as the defaults triggered here (so the current state of this solution is probably fine), but at least we can make sure we've been following whatever guidance has been bestowed to us by numpy / conda-forge.

**Comment by joaander** [comment]
@conda-forge-admin, please rerender

**Comment by IAlibay** [comment]
@conda-forge-admin, please restart ci

**Review by richardjgowers — APPROVED** [review]

**Review by IAlibay — APPROVED** [review]
lgtm thanks @joaander !

**Pull request merged**